### PR TITLE
Fix reversed bytes of (some) MAC commands

### PR DESCRIPTION
**Pull request by atifaziz**
# PR for issue #1281

## What is being addressed

Bytes of some MAC commands were reversed.

## How is this addressed

All MAC commands now have their bytes written in the natural order (first to last).

Other changes made to align some properties of MAC commands with actual types (avoids casts later):

- Property `Frequency` of `RXParamSetupRequest` was changed from `byte[]` to `int` (actually 24-bit and validated).
- Property `Freq` of `NewChannelRequest` was changed from `byte[]` to `int` (actually 24-bit and validated).
- Properties `Margin` and `Gwcnt` of `LinkCheckAnswer` were changed from `uint` to `byte`.

### CODE REVIEW

File: Tests/Integration/CloudToDeviceMessageSizeLimitShouldAcceptTests.cs [hunk 1]
```diff
@@ -141,7 +141,7 @@ public class CloudToDeviceMessageSizeLimitShouldAcceptTests : CloudToDeviceMessa
             if (expectedMacCommandsCount > 0)
             {
                 // Possible problem: Manually casting payloadDataDown.Fopts to array and reversing it
```
Comment (danigian) [inline Tests/Integration/CloudToDeviceMessageSizeLimitShouldAcceptTests.cs:143]: Remove the comment

**Review by danigian — COMMENTED** [review]
Looks great. Before approving, I would like to do some manual tests with real devices (even though I see the Mac E2E tests pass)

File: Tests/Integration/CloudToDeviceMessageSizeLimitShouldAcceptTests.cs (hunk 1 above)
Comment (atifaziz) [reply Tests/Integration/CloudToDeviceMessageSizeLimitShouldAcceptTests.cs]: Done with 4b057e1b4542dcc991592714f6fefe59a6577295.

**Review by danigian — APPROVED** [review]
E2E passed, plus I enjoyed seeing how flawless the test of LinkCheckReq, DutyCycleReq, DevStatusReq, RXTimingSetupReq mac commands was.

**Pull request merged**